### PR TITLE
INT-650 Adjust API for new renamed entities

### PR DESCRIPTION
**Pull request by Onbehalfofme**

### CODE REVIEW

File: talk/talk-api.yaml [hunk 1]
```diff
@@ -812,7 +812,7 @@ paths:
           application/x-www-form-urlencoded:
             schema:
               properties:
-                CallSid:
+                TwilioSid:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:815]: callbacks from Twilio, revert :)

File: talk/talk-api.yaml [hunk 2]
```diff
@@ -849,7 +849,7 @@ paths:
               properties:
                 To:
                   type: string
-                CallSid:
+                TwilioSid:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:852]: callbacks from Twilio, revert :)

File: talk/talk-api.yaml [hunk 3]
```diff
@@ -1770,13 +1771,36 @@ paths:
                 $ref: '#/components/schemas/TenantSettings'
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /tenant:
+    x-access: private
+    get:
+      tags:
+        - Tenant
+      x-access: private
+      summary: Retrieve tenant
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1780]:
```suggestion
      summary: Retrieve tenants
```

File: talk/talk-api.yaml [hunk 4]
```diff
@@ -1770,13 +1771,36 @@ paths:
...
+      description: THIS IS INTERNAL ENDPOINT. Use this endpoint to get settings configured on the tenant level. This operation supports paging and sorting.
+      operationId: getTenant
+      parameters:
+        - $ref: '#/components/parameters/QueryPageNumber'
+        - $ref: '#/components/parameters/QueryPageSize'
+        - $ref: '#/components/parameters/QuerySort'
+      responses:
+        200:
+          description: The tenant has been retrieved. Returns tenant.
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1789]:
```suggestion
          description: The tenants have been retrieved. Returns tenants.
```

File: talk/talk-api.yaml [hunk 5]
```diff
@@ -1841,6 +1902,43 @@ paths:
                 $ref: '#/components/schemas/UserSettings'
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /user:
+    get:
+      tags:
+        - User
+      summary: List user account
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1909]:
```suggestion
      summary: List user accounts
```

File: talk/talk-api.yaml [hunk 6]
```diff
@@ -1804,8 +1829,43 @@ paths:
           description: The spam blocking settings have been updated. Returns spam blocking settings.
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /tenant/spamBlocking:
+    get:
+      tags:
+        - Tenant
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1835]:
```suggestion
        - Tenants
```
all tags are plural

File: talk/talk-api.yaml [hunk 7]
```diff
@@ -1770,13 +1771,36 @@ paths:
                 $ref: '#/components/schemas/TenantSettings'
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /tenant:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1774]:
```suggestion
  /tenants:
```
all endpoints are plural too

File: talk/talk-api.yaml [hunk 8]
```diff
@@ -1841,6 +1902,43 @@ paths:
...
+      summary: List user account
+      description: Use this endpoint to retrieve account for a currently authenticated user.
+      operationId: getUser
+      responses:
+        200:
+          description: The user accounts have been retrieved. Returns user accounts.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserAccount'
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1918]: let's discuss that, but I think we have to generify /tenants and /users, so maybe the list should not be return in both of them

File: talk/talk-api.yaml [hunk 9]
```diff
@@ -2836,7 +2934,7 @@ components:
         - AccountSid
         - From
         - To
-        - CallSid
+        - TwilioSid
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:2937]: all twilio models should be reverted

File: talk/talk-api.yaml [hunk 10]
```diff
@@ -3015,7 +3113,7 @@ components:
         phoneNumber:
           type: string
           description: The full phone number including the country code.
-        webAgentId:
+        userId:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:3116]: deprecate the old one and add a new one

**Review by BullDog57Rus — CHANGES_REQUESTED** [review]

File: talk/talk-api.yaml (hunk 8 above)
Comment (Onbehalfofme) [reply talk/talk-api.yaml:1918]: I agree that we should get rid of UserAccounts, but then, if we assume that there is only one user or tenant we should leave descriptions and endpoints.

File: talk/talk-api.yaml [hunk 11]
```diff
@@ -1804,8 +1825,43 @@ paths:
           description: The spam blocking settings have been updated. Returns spam blocking settings.
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /tenants/spamBlocking:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1828]:
```suggestion
  /tenants/current/spamBlocking:
```

File: talk/talk-api.yaml [hunk 12]
```diff
@@ -1804,8 +1825,43 @@ paths:
...
+    get:
+      tags:
+        - Tenants
+      summary: List spam blocking settings
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1832]:
```suggestion
      summary: Retrieve current tenant's spam blocking settings
```

File: talk/talk-api.yaml [hunk 13]
```diff
@@ -1804,8 +1825,43 @@ paths:
...
+      description: Use this endpoint to retrieve spam blocking settings on the tenant level.
+      operationId: getSpamBlocking
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1834]:
```suggestion
      operationId: getCurrentSpamBlocking
```

File: talk/talk-api.yaml [hunk 14]
```diff
@@ -1804,8 +1825,43 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SpamBlockingSettings'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+    put:
+      tags:
+        - Tenants
+      summary: Update spam blocking settings
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1847]:
```suggestion
      summary: Update current tenant's spam blocking settings
```

File: talk/talk-api.yaml [hunk 15]
```diff
@@ -1804,8 +1825,43 @@ paths:
...
+      requestBody:
+        description: In the request body, provide new spam blocking settings.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SpamBlockingSettings'
+        required: true
+      responses:
+        200:
+          description: The spam blocking settings have been updated. Returns spam blocking settings.
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:1859]:
```suggestion
          description: The spam blocking settings have been updated. Returns spam blocking settings.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SpamBlockingSettings'
```

File: talk/talk-api.yaml [hunk 16]
```diff
@@ -3589,6 +3719,16 @@ components:
           $ref: '#/components/schemas/PhoneNumber'
         defaultCallForwarding:
           $ref: '#/components/schemas/CallForwarding'
+    UserAccount:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:3722]: just User is ok, I think

**Review by BullDog57Rus — CHANGES_REQUESTED** [review]

**Review by BullDog57Rus — APPROVED** [review]

**Comment by olyavertwist** [comment]
Guys, I have a question for you because I'm not sure what is the best solution here. I see that you mark some methods "deprecated". Do you want them to be visible in the API docs? If you want to hide them, please add 'x-access: private'. If they must be visible, please DM me & we'll figure out how to show them properly.

File: talk/talk-api.yaml [hunk 17]
```diff
@@ -1804,8 +1825,47 @@ paths:
           description: The spam blocking settings have been updated. Returns spam blocking settings.
         default:
           $ref: '#/components/responses/ErrorResponse'
+  /tenants/current/spamBlocking:
+    get:
+      tags:
+        - Tenants
+      summary: Retrieve current tenant's spam blocking settings
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1832]:
```suggestion
      summary: Retrieve spam blocking settings
```

File: talk/talk-api.yaml [hunk 18]
```diff
@@ -1804,8 +1825,47 @@ paths:
...
+      description: Use this endpoint to retrieve spam blocking settings on the tenant level.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1833]:
```suggestion
      description: Use this endpoint to retrieve spam blocking settings for a current tenant.
```

File: talk/talk-api.yaml [hunk 19]
```diff
@@ -1804,8 +1825,47 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SpamBlockingSettings'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+    put:
+      tags:
+        - Tenants
+      summary: Update current tenant's spam blocking settings
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1847]:
```suggestion
      summary: Update spam blocking settings
```

File: talk/talk-api.yaml [hunk 20]
```diff
@@ -1804,8 +1825,47 @@ paths:
...
+      description: Use this endpoint to update spam blocking settings configured on the tenant level.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1848]:
```suggestion
      description: Use this endpoint to update spam blocking settings for a current tenant.
```

File: talk/talk-api.yaml [hunk 21]
```diff
@@ -1831,14 +1892,51 @@ paths:
...
               schema:
-                $ref: '#/components/schemas/UserSettings'
+                $ref: '#/components/schemas/UserConfiguration'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+  /users/current:
+    get:
+      tags:
+        - Users
+      summary: Retrieve current user configuration.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1909]:
```suggestion
      summary: Retrieve configuration
```

File: talk/talk-api.yaml [hunk 22]
```diff
@@ -1831,14 +1892,51 @@ paths:
...
+      description: Use this endpoint to retrieve configuration for a currently authenticated user.
+      operationId: getCurrentUserConfiguration
+      responses:
+        200:
+          description: The current user configuration has been retrieved. Returns current user configuration.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1914]:
```suggestion
          description: The current user's configuration has been retrieved. Returns current user's configuration.
```

File: talk/talk-api.yaml [hunk 23]
```diff
@@ -1831,14 +1892,51 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UserConfiguration'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+    put:
+      tags:
+        - Users
+      summary: Update current user configuration
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1924]:
```suggestion
      summary: Update configuration
```

File: talk/talk-api.yaml [hunk 24]
```diff
@@ -1831,14 +1892,51 @@ paths:
...
+      description: Use this endpoint to modify and overwrite current user configuration.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1925]:
```suggestion
      description: Use this endpoint to modify and overwrite the current user's configuration.
```

File: talk/talk-api.yaml [hunk 25]
```diff
@@ -1831,14 +1892,51 @@ paths:
...
+      operationId: updateCurrentUserConfiguration
+      requestBody:
+        description: In the request body, pass the UserRequest object.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1928]:
```suggestion
        description: In the request body, pass the UserConfigurationRequest object.
```

**Review by olyavertwist — CHANGES_REQUESTED** [review]

**Review by nurutdinov — APPROVED** [review]

File: talk/talk-api.yaml [hunk 26]
```diff
@@ -1804,8 +1831,47 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SpamBlockingSettings'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+    put:
+      tags:
+        - Tenants
+      summary: Update current tenant's spam blocking settings
```
Comment (olyavertwist) [inline talk/talk-api.yaml:1853]:
```suggestion
        - Tenants
      summary: Update spam blocking settings
```

File: talk/talk-api.yaml [hunk 27]
```diff
@@ -66,12 +66,18 @@ tags:
...
     description: |
       Inperium Talk provides various communication services such as the ability to generate voice recordings based on text and vice versa.
-  - name: Tenant Settings
+  - name: Tenants
     description: |
       Inperium Talk enables users to get tenant settings.
-  - name: User Settings
+  - name: Tenant settings
+    description: |
+      Inperium Talk enables users to get tenant settings.(Deprecated)
```
Comment (olyavertwist) [inline talk/talk-api.yaml:74]:
```suggestion
      Currently, these endpoints are deprecated. Please leverage endpoints from the `Tenants` section.
```

File: talk/talk-api.yaml [hunk 28]
```diff
@@ -66,12 +66,18 @@ tags:
   - name: Services
     description: |
       Inperium Talk provides various communication services such as the ability to generate voice recordings based on text and vice versa.
-  - name: Tenant Settings
+  - name: Tenants
     description: |
       Inperium Talk enables users to get tenant settings.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:71]:
```suggestion
      Inperium Talk enables users to manage common tenant settings such as spam blocking.
```

File: talk/talk-api.yaml [hunk 29]
```diff
@@ -66,12 +66,18 @@ tags:
...
-  - name: User Settings
+  - name: Tenant settings
```
Comment (olyavertwist) [inline talk/talk-api.yaml:72]:
```suggestion
  - name: Tenant Settings
```

File: talk/talk-api.yaml [hunk 30]
```diff
@@ -66,12 +66,18 @@ tags:
...
+    description: |
+      Inperium Talk enables users to get tenant settings.(Deprecated)
+  - name: Users
     description: |
       Each user has a unique profile that contains information about the default contact number, forwarding options, and many more.
+  - name: User settings
+    description: |
+      Each user has a unique profile that contains information about the default contact number, forwarding options, and many more.(Deprecated)
```
Comment (olyavertwist) [inline talk/talk-api.yaml:80]:
```suggestion
  - name: User Settings
    description: |
      These endpoints are now deprecated. Please leverage endpoints from the 'Users' section.
```

**Review by olyavertwist — CHANGES_REQUESTED** [review]

**Review by olyavertwist — APPROVED** [review]

**Pull request merged**